src/org/labkey/test/tests/DataClassFolderExportImportTest.java

```diff
@@ -0,0 +1,351 @@
+        List<Map<String, String>> sourceRowData = new ArrayList<>();
+        for (int i=0; i<_attachments.size(); i++)
+        {
+            sourceTable.clickEditRow(i);
+            setFormElement(Locator.input("quf_attachmentColumn"), _attachments.get(i));
+            clickButton("Submit");
+        }
+        for (int i=0; i<expectedRecordCount; i++)
+        {
+            sourceRowData.add(sourceTable.getRowDataAsMap(i));
+        }
+
+        // act - export to file and import the file to our designated import directory
+        goToFolderManagement()
+                .goToExportTab();
+        File exportedFolderFile = doAndWaitForDownload(()->findButton("Export").click());
+
+        goToProjectFolder(IMPORT_PROJECT_NAME, importFolder);
+        importFolderFromZip(exportedFolderFile, false, 1);
+        goToProjectFolder(IMPORT_PROJECT_NAME, importFolder);
+
+        List<String> importedRunNames = DataRegionTable.DataRegion(getDriver()).withName("Runs").waitFor()
+                .getColumnDataAsText("Name");
+        for (String sourceRun : runNames)
+        {
+            assertThat("expect all runs to come through", importedRunNames, hasItems(sourceRun));
+        }
+
+        // we expect the sample types and experiment runs webparts to come across along with the folder import
+        clickAndWait(Locator.linkWithText(testDataClass));
+        DataRegionTable destSamplesTable = DataRegionTable.DataRegion(getDriver()).withName("query").waitFor();;
+
+        // capture the data in the exported sampleType
+        List<Map<String, String>> destRowData = new ArrayList<>();
+        for (int i=0; i<expectedRecordCount; i++)
+        {
+            destRowData.add(destSamplesTable.getRowDataAsMap(i));
+        }
+
+        // now ensure expected data in the sampleType made it to the destination folder
+        for (Map exportedRow : sourceRowData)
+        {
+            // find the map from the exported project with the same name
+            Map<String, String> matchingMap = destRowData.stream().filter(a-> a.get("Name").equals(exportedRow.get("Name")))
+                    .findFirst().orElse(null);
+            assertNotNull("expect all matching rows to come through", matchingMap);
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("intColumn"), equalTo(matchingMap.get("intColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("stringColumn"), equalTo(matchingMap.get("stringColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("decimalColumn"), equalTo(matchingMap.get("decimalColumn")));
+            assertThat("expect export and import values to be equivalent",
+                    exportedRow.get("attachmentColumn"), equalTo(matchingMap.get("attachmentColumn")));
+        }
```

Comment:
I don't think we need to go through all this rigamarole to verify the data. `assertEquals` works pretty well on collections.
```suggestion
        // now ensure expected data in the sampleType made it to the destination folder
        assertEquals("Data class did not match after round tripping.", sourceRowData, destRowData);
```

Reply:
done